### DISCUSSION TITLE
abstract final class for freezed class

### DISCUSSION THREAD

**Discussion by leobellejoseph** [rrousselGit/freezed#1123] [General]
Hi,

I would like to ask for advise. currently i am implementing the dcm rule prefer-abstract-final-static-class in which this will make me implement my freezed class to be abstract final class. however i encounter an issue on the generated class. I would like to seek advise if for freezed class this rule should be ignored? thank you.

**Comment by rrousselGit** [comment]
You can't use the `final` keyword here, because the abstract class will be subclassed in generated code.

Try using `sealed`.

**Reply by leobellejoseph** [reply]
thank you very much @rrousselGit
